zerver/views/public_export.py

```diff
@@ -0,0 +1,52 @@
+import os
+import shutil
+import tempfile
+import datetime
+
+from django.utils.timezone import now as timezone_now
+from django.utils.translation import ugettext as _
+
+from zerver.models import RealmAuditLog, Realm
+from zerver.lib.export_realm import command_independent_realm_export
+from zerver.lib.queue import queue_json_publish
+from zerver.lib.response import json_error
+from zerver.tornado.event_queue import send_event
+
+def public_only_realm_export(realm: Realm) -> None:
+    event_type = RealmAuditLog.REALM_EXPORT_TRIGGERED
+    event_time = timezone_now()
+    # Check for event times occurring in the past week
+    event_time_delta = event_time - datetime.timedelta(days=7)
+
+    # A basic implementation for rate-limiting realm exports
+    # We check that the number of events taking place within our delta does not exceed the limit
+    realm_exports = RealmAuditLog.objects.filter(realm=realm, event_type=event_type)
+    realm_exports = [log for log in realm_exports if event_time > event_time_delta]
```

Comment:
Shouldn't this by log.event_time?
Also this won't be required if we do it the way I described.